builder/googlecompute/config.go

```diff
@@ -195,6 +195,13 @@ type Config struct {
 	// The path to a startup script to run on the VM from which the image will
 	// be made.
 	StartupScriptFile string `mapstructure:"startup_script_file" required:"false"`
+	// If "true", the contents of `startup_script_file` is wrapped in a Packer specific
+	// script that tracks the execution and completion of the provided
+	// script file. The wrapper ensures that the builder will not continue until
+	// the startup script has been executed. Defaults to "true".
+	// - The use of the wrapped script file requires that the user or service account
+	// running the build has the compute.instance.Metadata role.
+	WrapStartupScriptFile config.Trilean `mapstructure:"wrap_startup_script" required:"false"`
```

Comment:
Should this be a simple boolean ? Trilean is neat for when you want three specific behaviors, true, false or unset. 
```suggestion
	WrapStartupScriptFile bool `mapstructure:"wrap_startup_script" required:"false"`
```

Reply:
I understood that the `unset` behavior is important to know so it could set the default do `true`.

Reply:
> I understood that the unset behavior is important to know so it could set the default do true.

@sylviamoss that's correct. I started with a boolean here but the default case should be true until we completely deprecated the wrap script. So I went with the Trilean because the unset state is important here.

Reply:
If the only issue is being able to default true instead of false, I think we can use mapstructure to do that: https://github.com/hashicorp/packer/blob/master/post-processor/vsphere-template/post-processor.go#L45

Though it's worth verifying that this defaults as we expect.

Reply:
Ah yes I see, default to true, a solution would be to negate the name of the variable so the golang's 'default to false for bool behaviour' just gives you this:
```suggestion
	UnwrapStartupScriptFile bool `mapstructure:"wrap_startup_script" required:"false"`
```

Reply:
I think negating the name makes it harder to conceptualize, and also when we switch this to defaulting to false it'll be the same issue. Let's go with the original Trilean solution.

Reply:
Got it! I reverted back to the original implementation and dropped a few updates to the documentation.